src/svgs/runin/runin-shmetro.tsx

```diff
@@ -235,17 +329,153 @@ const NextText = (props: { nextName: Name } & React.SVGProps<SVGGElement>) => {
+                    </>
+                ),
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                undefined
+            )}
+        </g>
+    );
+};
+
+const StationNumberPast = (props: { xpos: number, ypos: number, lineName: string, stationNumber: string } & React.SVGProps<SVGGElement>) => {
+    const { xpos, ypos, lineName, stationNumber, ...others } = props;
+    return (
+        <g >
+            {useMemo(
+                () => (
+                    <>
+                        <circle cx={xpos} cy={ypos} r="35" stroke="var(--rmg-grey)" strokeWidth="8" fill="white">
+                        </circle>
+                        <line x1={xpos - 30} y1={ypos} x2={xpos + 30} y2={ypos} stroke="rgb(0,0,0)" strokeWidth="2" />
+                        <StationNumberText transform={"translate(" + (xpos).toString() + "," + (ypos - 5).toString() + ")"} lineNum={lineName} stationNum={stationNumber} />
+                    </>
+                ),
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                undefined
+            )}
+        </g>
+    );
+};
+
+const StationNumberNow = (props: { xpos: number, ypos: number, lineName: string, stationNumber: string } & React.SVGProps<SVGGElement>) => {
+    const { xpos, ypos, lineName, stationNumber, ...others } = props;
+    return (
+        <g >
+            {useMemo(
+                () => (
+                    <>
+                        <circle cx={xpos} cy={ypos} r="45" stroke="var(--rmg-theme-colour)" strokeWidth="10" fill="white">
+                        </circle>
+                        <line x1={xpos - 40} y1={ypos} x2={xpos + 40} y2={ypos} stroke="rgb(0,0,0)" strokeWidth="2" />
+                        <StationNumberTextNow transform={"translate(" + (xpos).toString() + "," + (ypos - 5).toString() + ")"} lineNum={lineName} stationNum={stationNumber} />
+                    </>
+                ),
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                undefined
+            )}
+        </g>
+    );
+};
 
 const PrevStn = (props: { stnIds: string[] }) => {
     const param = useAppSelector(store => store.param);
     const nextNames = props.stnIds.map(stnId => param.stn_list[stnId].name);
+    const nextNumbers = props.stnIds.map(stnId => param.stn_list[stnId].num);
     const prevHintDy = (props.stnIds.length > 1 ? 15 : 125) +
         nextNames.map(name => name[0].split('\\').length).reduce((acc, cur) => acc + cur, -nextNames.length) * -50 +
         nextNames.map(name => name[1].split('\\').length).reduce((acc, cur) => acc + cur, -nextNames.length) * -30
     const nextBranchTextDy = (props.stnIds.length > 1 ? (nextNames[0][0].split('\\').length - 1) * -50 +
         (nextNames[0][1].split('\\').length - 1) * -30 : 0) + 70
 
-    return (
+    return (param.stn_number === undefined ? false : param.stn_number) ? (
```

Comment:
The same as above. You may try {param.stn_number === true && <StationNumberPast>...</<StationNumberPast>} to git rid of large code duplication.